tutorials/xr/openxr_spatial_entities.rst

```diff
@@ -0,0 +1,1736 @@
+
+                # Remove the entity from our original list
+                if org_entities.has(entity_id):
+                    org_entities.erase(entity_id)
+
+                if entity_state == OpenXRSpatialEntityTracker.ENTITY_TRACKING_STATE_STOPPED:
+                    # We're not doing update snapshots so we shouldn't get this,
+                    # but just to future proof:
+                    if entities.has(entity_id):
+                        var entity_tracker : OpenXRMarkerTracker = entities[entity_id]
+                        entity_tracker.spatial_tracking_state = entity_state
+                        XRServer.remove_tracker(entity_tracker)
+                        entities.erase(entity_id)
+                else:
+                    var entity_tracker : OpenXRMarkerTracker
+                    var register_with_xr_server : bool = false
+                    if entities.has(entity_id):
+                        entity_tracker = entities[entity_id]
+                    else:
+                        entity_tracker = OpenXRMarkerTracker.new()
+                        entity_tracker.entity = OpenXRSpatialEntityExtension.make_spatial_entity(spatial_context, entity_id)
+                        entities[entity_id] = entity_tracker
+                        register_with_xr_server = true
+
+                    # Copy the state.
+                    entity_tracker.spatial_tracking_state = entity_state
+
+                    # If we're tracking, we should query the rest of our components.
+                    if entity_state == OpenXRSpatialEntityTracker.ENTITY_TRACKING_STATE_TRACKING:
+                        var center_pose : Transform3D = bounded2d_list.get_center_pose(i)
+                        entity_tracker.set_pose("default", center_pose, Vector3(), Vector3(), XRPose.XR_TRACKING_CONFIDENCE_HIGH)
+
+                        entity_tracker.bounds_size = bounded2d_list.get_size(i)
+
+                        entity_tracker.marker_type = marker_list.get_marker_type(i)
+                        entity_tracker.marker_id = marker_list.get_marker_id(i)
+                        entity_tracker.marker_data = marker_list.get_marker_data(p_snapshot, i)
+                    else:
+                        entity_tracker.invalidate_pose("default")
+
+                    # We don't register our tracker until after we've set our initial data.
+                    if register_with_xr_server:
+                        XRServer.add_tracker(entity_tracker)
+
+        # Any entities we've got left over, we can remove.
+        for entity_id in org_entities:
+            var entity_tracker : OpenXRMarkerTracker = entities[entity_id]
+            entity_tracker.spatial_tracking_state = OpenXRSpatialEntityTracker.ENTITY_TRACKING_STATE_STOPPED
+            XRServer.remove_tracker(entity_tracker)
+            entities.erase(entity_id)
+
+
+
+
+    **
+        Q : Should we be doing update queries here to get position changes for markers??
```

Comment:
Mostly a reminder to myself, I need to add this in. We now do this in the built-in logic as well.